numpy/core/src/multiarray/methods.c

```diff
@@ -1660,13 +1660,15 @@ array_reduce_ex(PyArrayObject *self, PyObject *args)
             /* if the python version is below 3.8, the pickle module does not provide
              * built-in support for protocol 5. We try importing the pickle5
              * backport instead */
-            if (PY_VERSION_HEX < 0x03080000){
-                pickle_module = PyImport_ImportModule("pickle5");
-            }
-
-            else {
-                pickle_module = PyImport_ImportModule("pickle");
-            }
+#if defined(NPY_PY3K)
+#if PY_VERSION_HEX < 0x03080000 && PY_VERSION_HEX >= 0x03070000
+            pickle_module = PyImport_ImportModule("pickle5");
+#else
+            pickle_module = PyImport_ImportModule("pickle");
+#endif
+#else
+            pickle_module = PyImport_ImportModule("cPickle");
+#endif
```

Comment:
[pickle5](https://pypi.org/project/pickle5/) works for both Python 3.6 and 3.7. So:

- under Python 3.8, the pickle module should always work;
- under Python 3.6 and 3.7 we should try to import `pickle5` and raise `ImportError` if not installed.
- under Python <= 3.5: we should always raise a `ValueError` stating that `protocol=5` is invalid for old Python versions.

 So if `__reduce_ex__` is called with `protocol=5`, and pickle5 is not installed I think we should raise